zipline/data/data_portal.py

```diff
@@ -773,19 +773,20 @@ def get_spot_value(self, asset, field, dt, data_frequency):
         if field not in BASE_FIELDS:
             raise KeyError("Invalid column: " + str(field))
 
+        session_label = self.trading_calendar.minute_to_session_label(dt)
```

Comment:
I was hoping we wouldn't have to do this translation in multiple places, but to abstract it away in one location.  Any chance that's possible?